### ISSUE TITLE
Fails to install Aftermath data

### ISSUE THREAD

**Issue by paulvt** [OpenRA/OpenRA#12009]
When trying to install the data from the Aftermath data/ISO, an error occurs.  The log reports the following:

```
Extracting /media/cdrom0/main.mix -> /home/paul/.openra/Content/ra/v2/expand/2nd_hand.aud
Extracting /media/cdrom0/main.mix -> /home/paul/.openra/Content/ra/v2/expand/araziod.aud
Extracting /media/cdrom0/main.mix -> /home/paul/.openra/Content/ra/v2/expand/backstab.aud
Extracting /media/cdrom0/main.mix -> /home/paul/.openra/Content/ra/v2/expand/chaos2.aud
Extracting /media/cdrom0/main.mix -> /home/paul/.openra/Content/ra/v2/expand/shut_it.aud
Extracting /media/cdrom0/main.mix -> /home/paul/.openra/Content/ra/v2/expand/twinmix1.aud
Extracting /media/cdrom0/main.mix -> /home/paul/.openra/Content/ra/v2/expand/under3.aud
Extracting /media/cdrom0/main.mix -> /home/paul/.openra/Content/ra/v2/expand/vr2.aud
System.IO.DirectoryNotFoundException: Could not find a part of the path "/usr/lib/openra/SETUP/
INSTALL/PATCH.RTP".
  at System.IO.FileStream..ctor (System.String path, FileMode mode, FileAccess access, FileShar
e share, Int32 bufferSize, Boolean anonymous, FileOptions options) <0x7fdf2f8682c0 + 0x001f4> i
n <filename unknown>:0 
  at System.IO.FileStream..ctor (System.String path, FileMode mode, FileAccess access, FileShare share) <0x7fdf2f867f30 + 0x0004b> in <filename unknown>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileStream:.ctor (string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare)
  at System.IO.File.OpenRead (System.String path) <0x7fdf2f865a50 + 0x00034> in <filename unknown>:0 
  at OpenRA.Mods.Common.Widgets.Logic.InstallFromDiscLogic.ExtractFromPackage (ExtractionType type, System.String path, OpenRA.MiniYaml actionYaml, System.Collections.Generic.List`1 extractedFiles, System.Action`1 updateMessage) <0x417a3f90 + 0x00126> in <filename unknown>:0 
  at OpenRA.Mods.Common.Widgets.Logic.InstallFromDiscLogic+<InstallFromDisc>c__AnonStorey5.<>m__1 () <0x417a2f90 + 0x007eb> in <filename unknown>:0 
System.IO.DirectoryNotFoundException: Could not find a part of the path "/usr/lib/openra/SETUP/INSTALL/PATCH.RTP".
  at System.IO.FileStream..ctor (System.String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, Boolean anonymous, FileOptions options) <0x7fdf2f8682c0 + 0x001f4> in <filename unknown>:0 
  at System.IO.FileStream..ctor (System.String path, FileMode mode, FileAccess access, FileShare share) <0x7fdf2f867f30 + 0x0004b> in <filename unknown>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileStream:.ctor (string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare)
  at System.IO.File.OpenRead (System.String path) <0x7fdf2f865a50 + 0x00034> in <filename unknown>:0 
  at OpenRA.Mods.Common.Widgets.Logic.InstallFromDiscLogic.ExtractFromPackage (ExtractionType type, System.String path, OpenRA.MiniYaml actionYaml, System.Collections.Generic.List`1 extractedFiles, System.Action`1 updateMessage) <0x417a3f90 + 0x00126> in <filename unknown>:0 
  at OpenRA.Mods.Common.Widgets.Logic.InstallFromDiscLogic+<InstallFromDisc>c__AnonStorey5.<>m__1 () <0x417a2f90 + 0x007eb> in <filename unknown>:0 
```

The path seems to be taken relative to the install location of OpenRA instead of the CD location.

**Comment by pchote** [comment]
Works for me on OSX, so this isn't as simple as it using the install location instead of the disc.

**Comment by paulvt** [comment]
I would be nice if we could reproduce and debug thus.  Unfortunately it seems to have been an issue with the ISO and it no longer exists.  Closing it until someone encounters (possibly) it again.

**Comment by docholiday** [comment]
Not sure, if this is the same problem, but the Windows Installer (I am running Win 7 64Bit) fails to recognize the Aftermath ISO (from cncnz.com). See my issue [https://github.com/OpenRA/OpenRAWeb/issues/329](url)

**Comment by pchote** [comment]
As far as I can tell the iso from CNCNZ is corrupted.  It won't even mount on macOS.

**Comment by docholiday** [comment]
Hmm, mounting worked flawlessly, and the files looked ok, but I'll try another one, thanks!

**Comment by docholiday** [comment]
Found another image that worked, thanks again :-)

**Comment by Walkman100** [comment]
This is still a bug, the Aftermath iso from http://www.cncnz.com/features/freeware-classic-command-conquer-games/ on OpenRA-release-20170527 still gives this error:
```
System.IO.DirectoryNotFoundException: Could not find a part of the path "/usr/lib/openra/SETUP/INSTALL/PATCH.RTP".
  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share, System.Int32 bufferSize, System.Boolean anonymous, System.IO.FileOptions options) [0x00164] in <204f770036d441bb8dfd3daba3550e83>:0 
  at System.IO.FileStream..ctor (System.String path, System.IO.FileMode mode, System.IO.FileAccess access, System.IO.FileShare share) [0x00000] in <204f770036d441bb8dfd3daba3550e83>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileStream:.ctor (string,System.IO.FileMode,System.IO.FileAccess,System.IO.FileShare)
  at System.IO.File.OpenRead (System.String path) [0x00000] in <204f770036d441bb8dfd3daba3550e83>:0 
  at OpenRA.Mods.Common.Widgets.Logic.InstallFromDiscLogic.ExtractFromPackage (OpenRA.Mods.Common.Widgets.Logic.InstallFromDiscLogic+ExtractionType type, System.String path, OpenRA.MiniYaml actionYaml, System.Collections.Generic.List`1[T] extractedFiles, System.Action`1[T] updateMessage) [0x00033] in <fdc1a2308ba047f88b49109de7f2737a>:0 
  at OpenRA.Mods.Common.Widgets.Logic.InstallFromDiscLogic+<InstallFromDisc>c__AnonStorey5.<>m__1 () [0x00238] in <fdc1a2308ba047f88b49109de7f2737a>:0 
```
However the files exist in the mounted filesystem, and if I copy the files from the mounted disc (as root) to /usr/lib/openra and rename the mentioned file to uppercase, then the music gets extracted correctly:
![image](https://user-images.githubusercontent.com/5454764/29698795-8ee8124c-8958-11e7-905b-fb79923d7df7.png)

Interestingly my archive manager (ark) reads the ISO as empty, but a simple `7z l` on it reveals all the contents uncorrupted.

**Comment by pchote** [comment]
This was fixed by #13760 which will be shipped with the next playtest.
